presto-verifier/src/main/java/com/facebook/presto/verifier/checksum/ChecksumValidator.java

```diff
@@ -87,6 +87,27 @@ public Query generatePartitionChecksumQuery(QualifiedName tableName, List<Column
                 Optional.empty());
     }
 
+    public Query generateBucketChecksumQuery(QualifiedName tableName, List<Column> columns)
+    {
+        ImmutableList.Builder<SelectItem> selectItems = ImmutableList.builder();
+        selectItems.add(new SingleColumn(new FunctionCall(QualifiedName.of("count"), ImmutableList.of())));
+        for (Column column : columns) {
+            selectItems.addAll(columnValidators.get(column.getCategory()).get().generateChecksumColumns(column));
+        }
+
+        ImmutableList.Builder<GroupingElement> groupByList = ImmutableList.builder();
+        groupByList.add(new SimpleGroupBy(ImmutableList.of(new Identifier("$bucket"))));
```

Comment:
I think for bucket, we compute checksum per partition per bucket. Here, if I look at the right place, it looks we compute checksum per bucket across all partitions, which might not be quite useful. Immc we might have touched this in your design meeting. But you can double check with Masha.